grails-datastore-gorm/src/main/groovy/grails/gorm/multitenancy/Tenants.groovy

```diff
@@ -232,7 +271,7 @@ class Tenants {
     * @param callable The closure
     * @return The result of the closure
     */
-    static <T> T withId(MultiTenantCapableDatastore multiTenantCapableDatastore, Serializable tenantId, Closure<T> callable) {
+    static <T> T withId(MultiTenantCapableDatastore multiTenantCapableDatastore, Serializable tenantId, Closure callable) {
```

Comment:
Any reason why the Generics on the `Closure callable` parameter was dropped. That would be the return value type (`<T>`)

Reply:
I believe this was because there's some odd type erasure going on with groovy 4: 

    grails-datastore-gorm-rx\build\tmp\compileGroovy\groovy-java-stubs\grails\gorm\rx\multitenancy\Tenants.java:11: error: name clash: <T#1>withCurrent(Class<? extends RxDatastoreClient>,Closure<T#1>) in grails.gorm.rx.multitenancy.Tenants and <T#2>withCurrent(Class<? extends Datastore>,Closure<T#2>) in grails.gorm.multitenancy.Tenants have the same erasure, yet neither hides the other
    public static <T> T withCurrent(java.lang.Class<? extends org.grails.datastore.rx.RxDatastoreClient> datastoreClass, @groovy.lang.DelegatesTo(value=org.grails.datastore.rx.RxDatastoreClient.class) groovy.lang.Closure<T> callable) { return (T)null;}
                        ^
      where T#1,T#2 are type-variables:
        T#1 extends Object declared in method <T#1>withCurrent(Class<? extends RxDatastoreClient>,Closure<T#1>)
        T#2 extends Object declared in method <T#2>withCurrent(Class<? extends Datastore>,Closure<T#2>)
    Note: Some input files use unchecked or unsafe operations.
    Note: Recompile with -Xlint:unchecked for details.

Reply:
It was one of many attempts to work through issues with Tenants.groovy and Groovy 4.0.22  I think that class needs attacked from scratch by someone more knowledgeable about the erasure issue with Closures as a method parameter.